sql/src/main/java/io/crate/execution/dml/upsert/ShardUpsertRequest.java

```diff
@@ -231,35 +181,76 @@ public void writeTo(StreamOutput out) throws IOException {
         }
     }
 
-    @Override
-    protected Item readItem(StreamInput input) throws IOException {
-        return new Item(input, insertValuesStreamer);
+    @Nullable
+    public SessionSettings sessionSettings() {
+        return sessionSettings;
+    }
+
+    @Nullable
+    public Symbol[] returnValues() {
+        return returnValues;
+    }
+
+    @Nullable
+    public String[] updateColumns() {
+        return updateColumns;
+    }
+
+    @Nullable
+    public Reference[] insertColumns() {
+        return insertColumns;
+    }
+
+    public boolean continueOnError() {
+        return continueOnError;
+    }
+
+    public boolean validateConstraints() {
+        return validateConstraints;
+    }
+
+    public DuplicateKeyAction duplicateKeyAction() {
+        return duplicateKeyAction;
     }
 
     @Override
     public boolean equals(Object o) {
```

Comment:
ShardUpsertRequest.Item should have equals/hashcode since the superclass has it as well, so i regenerated it ShardUpsertRequest to be consistent and not use `import com.google.common.base.Objects` anymore.